isis/src/base/apps/getsn/getsn.cpp

```diff
@@ -0,0 +1,94 @@
+#include "getsn.h"
+
+#include <iostream>
+#include <sstream>
+#include <QString>
+
+#include "Blob.h"
+#include "Cube.h"
+#include "History.h"
+#include "ObservationNumber.h"
+#include "Preference.h"
+#include "Pvl.h"
+#include "SerialNumber.h"
+#include "SessionLog.h"
+#include "TextFile.h"
+
+using namespace Isis;
+using namespace std;
+
+
+namespace Isis {
+
+  void getsn( UserInterface &ui, Pvl *log ) {
+    // Open the input cube
+    Cube *cube = new Cube( ui.GetFileName("FROM"), "r");
+    
+    getsn( cube, ui, log );
+
+  }
+
+
+  void getsn( Cube *cube, UserInterface &ui, Pvl *log ) {
+
+    // Determine if output should be written base on parameters
+    bool WriteFile = ui.GetBoolean("FILE");
+    bool WriteSN = ui.GetBoolean("SN");
+    bool WriteObservation = ui.GetBoolean("OBSERVATION");
+
+    QString from = ui.GetFileName("FROM");    
```

Comment:
Repalce this with `QString from = cube-fileName()` otherwise the function will fail if the parameter doesn't exist.